### DISCUSSION TITLE
Sensors in HA

### DISCUSSION THREAD

**Discussion by Schalmeier** [egguy/grott-home-assistant-add-on#22] [General]
Hi,
i already had grott in a Docker instance running. Now i moved it to my HA with your addon. 
My motivation was, that i now don't have to manually add all the data trough MQTT to my HA and get some kind of Device information like with the flaky Growatt-intergration. But in my case i gained nothing with moving grott to HA, right? I mean, the addon doesn't generate any entities in HA. Or do i miss something? 

Thanks :)

**Comment by egguy** [comment]
Hello,

If you had the grott + grott HA addon in a docker. You won't get any new sensors, just auto-updates and bugfixes. This is basically a pre-packaged + pre/auto configured version of grott + grott HA addon who auto-connect to the MQTT of HA and a pretty config interface. 

If you were exporting the metrics to HA via the mqtt output of grott + YAML config, then you can get a lot of additional metrics autoconfigured (batteries, import/export, integration with the energy panel)+ multi inverter (excepted if you have done all these configurations in the YAML file).

e.g.
![image](https://user-images.githubusercontent.com/186777/232471410-36c7da78-4c26-44fd-b790-f70ab32015fe.png)
![image](https://user-images.githubusercontent.com/186777/232471513-e38a485e-ea36-4503-96fd-1645d0466d13.png)

**Reply by Schalmeier** [reply]
I export the data to a MQTT broker outside of HA. HA has access to it with the MQTT integration. So i need to manually configure all sensors in the configuration.yaml?
![Bild_2023-04-17_134333240](https://user-images.githubusercontent.com/126117846/232474543-cb985c98-8ce2-4b4a-9e67-3456d72ef8b7.png)

**Reply by egguy** [reply]
Normally, there shouldn't be a difference. Are you using the edge version ?
I think in the `protokoll` panel you should see if the add-on can connect and the logger communicate.

Also if the logger has been disconnected for a long time, it can take a bit of time until the data appears.

**Reply by Schalmeier** [reply]
There is no information which version i'm using... 
![Bild_2023-04-17_140110099](https://user-images.githubusercontent.com/126117846/232478200-b434238b-26ec-463a-92d1-42fefde42973.png)

I'm getting data in mqtt though

![Bild_2023-04-17_141340881](https://user-images.githubusercontent.com/126117846/232480798-cbc02162-f760-4a7a-8eb3-4682350e4012.png)
![image](https://user-images.githubusercontent.com/126117846/232480981-4dd84e41-6117-4c08-b523-79bdf517a5b5.png)

do i have to configure something, that HA automatically discovers this as device?

**Reply by egguy** [reply]
It seems you are not using the -edge version. I see the problem. I will try to correct it this weekend.

The most up-to-date version is on the right: 
![image](https://user-images.githubusercontent.com/186777/232503007-d91ec377-e8e1-4a75-bc92-fc779db25250.png)

I renamed and updated the description, but forgot to add the mention of edge version.

**Reply by Schalmeier** [reply]
> If you were exporting the metrics to HA via the mqtt output of grott + YAML config, then you can get a lot of additional metrics autoconfigured (batteries, import/export, integration with the energy panel)+ multi inverter (excepted if you have done all these configurations in the YAML file).
> 

I'm so sorry, but i don't get what you mean with that. 
Do i have to configure the each sensor in the YAML like any other MQTT Sensor? If so, could you please share your config, to copy these sensors from?
Or what do you mean by autoconfigured? When should this happen? Should the be auto discovered via the MQTT integration?

**Reply by egguy** [reply]
No you don't have to configure each sensor, if the grott HA extension is correctly configured (You should see information about creating  sensors the first time you receive a data packet from the logger) the sensors should appear automatically. 

In your last messages, It seems you were missing the MQTT configuration in the addon, you need to configure this manually when using an external broker.

The sensors should look like this in the MQTT broker if grott_ha extension is correctly configured: 
![image](https://user-images.githubusercontent.com/186777/232539858-6a122c9d-6a72-4cf7-83b6-46353510062c.png)
there is 69 topic one for each sensor with their own configuration.

But if you don't use/want the extension, and you are sending the MQTT message to HA **without the grott HA extension** (like in your capture), you have to follow something like this: https://github.com/johanmeijer/grott/wiki/Home-Assistant-configuration-examples

**Reply by Schalmeier** [reply]
Okay. Got it. Thanks for your patience! 

Now, i think, i found a bug (?) or my config is just wrong... 
This is the protocol from the extension. It tries to publish data to HA, but some kind of username variable is missing. 
I had pretty much the same error with the MQTT publish before, because it always expects a username for the MQTT Auth. This was resolved with emty config entries. 

![image](https://user-images.githubusercontent.com/126117846/232698099-7a8c8c94-625c-4dc2-9d3f-1522bc76b0c8.png)

But i can't find any missing config point here:

![image](https://user-images.githubusercontent.com/126117846/232699190-309e2b22-8755-4fb9-a618-e20dc5693a72.png)

**Reply by egguy** [reply]
It's an error in the documentation. I will rectify this. The good key is: `username`

**Comment by egguy** [comment]
Also you can add the inverter type parameter if your inverter is one of them:

- TL3-X 3 -> tl3
- SPF -> spf 
- SPH -> sph

This will display additional metrics in HA
